lib/api/rates.ts

```
@@ -1,11 +1,56 @@
+      .catch((e) => {
+        if (cancelled) return;
+        setData(null);
+        setError(e instanceof Error ? e.message : 'Failed to load rates');
+      })
+      .finally(() => {
+        if (!cancelled) setLoading(false);
+      });
+    return () => {
+      cancelled = true;
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

ESLint

[error] Definition for rule 'react-hooks/exhaustive-deps' was not found.

(react-hooks/exhaustive-deps)